packages/server/src/rooms.ts

```diff
@@ -33,26 +33,26 @@ export async function joinRoom(io: IOServer, socket: Socket, store: Store, roomN
 
   const player = room.players.find((player) => player.name === username);
   if (player) {
-    await doJoinRoom(socket, store, room, player);
-  } else if (room.players.length >= MAX_ROOM_SIZE) {
+    await doJoinRoom(socket, store, room);
+  } else if (room.players.length > MAX_ROOM_SIZE) {
     console.warn(`[JOIN FAILED] Room ${roomName} is full`);
     socket.emit(RoomEvent.JoinError, 'Room is full');
   } else {
     const newPlayer = new Player(username);
     store.addPlayer(roomName, newPlayer);
 
-    await doJoinRoom(socket, store, room, newPlayer);
+    await doJoinRoom(socket, store, room);
   }
 }
 
-async function doJoinRoom(socket: Socket, store: Store, room: Room, player: Player) {
+async function doJoinRoom(socket: Socket, store: Store, room: Room) {
```

Comment:
You should not just remove this... I see what is happening here, but I don't know how JS handles this. In line 58, I `player.name` is used from arrow function, or function parameter? In any case, parameter should just be `playerName`, since that's the only thing used from `player.

Reply:
I see it's used from the function parameter then, but I bet it's the same in python or any other languages, is that the issue? Let me try